Eugene/MovieSearch/my-app/src/components/MainPage/sort-panel/SortPanel.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import "./SortPanel.css";
+import { Location } from "history";
+import { parse } from "querystring";
+
+interface ISortPanelProps {
+  movieCount: number;
+  handleSort(sortBy: string): void;
+  location: Location;
+}
+
+interface ISortPanelState {
+  sortBy: string;
+}
+
+enum SortBy {
+  Rating = "vote_average",
+  ReleaseDate = "release_date",
+}
+
+class SortPanel extends React.Component<ISortPanelProps, ISortPanelState> {
+  state: ISortPanelState = {
+    sortBy: SortBy.Rating,
+  };
```

Comment:
You might not need state here. There is a difference between SearchPanel and SortPanel. In SearchPanel we change an URL only after submitting form, while in SortPanel we do it right away. That's why you probably don't need state, url IS your state. You can get rid of state and didMount and just set active sort from the url in props.